apps/judicial-system/backend/src/app/modules/notification/notification.service.ts

```diff
@@ -1224,39 +1245,67 @@ export class NotificationService {
   private async sendDefenderAssignedNotifications(
     theCase: Case,
   ): Promise<SendNotificationResponse> {
-    const defenderEmail = theCase.defenderEmail
-    if (!theCase.defenderNationalId || !defenderEmail) {
-      return Promise.resolve({ notificationSent: false })
-    }
-
+    const promises: Promise<Recipient>[] = []
     const pastNotifications = await this.notificationModel.findAll({
       where: { caseId: theCase.id, type: NotificationType.DEFENDER_ASSIGNED },
     })
-    const hasSentNotificationBefore = pastNotifications.some(({ recipients }) =>
-      recipients?.includes(defenderEmail),
-    )
 
-    if (hasSentNotificationBefore) {
-      return Promise.resolve({ notificationSent: false })
-    }
+    if (isIndictmentCase(theCase.type) && theCase.defendants) {
+      theCase.defendants.forEach((defendant) => {
+        const { defenderEmail, defenderNationalId, defenderName } = defendant
 
-    const { subject, body } = formatDefenderAssignedEmailNotification(
-      this.formatMessage,
-      theCase,
-      `${this.config.clientUrl}${DEFENDER_ROUTE}/${theCase.id}`,
-    )
+        if (!defenderNationalId || !defenderEmail) {
+          return Promise.resolve({ notificationSent: false })
+        }
 
-    const recipient = await this.sendEmail(
-      subject,
-      body,
-      theCase.defenderName,
-      defenderEmail,
-    )
+        const hasSentNotificationBefore = pastNotifications.some(
+          ({ recipients }) => recipients?.includes(defenderEmail),
+        )
+
+        if (hasSentNotificationBefore) {
+          return Promise.resolve({ notificationSent: false })
+        }
+
+        const { subject, body } = formatDefenderAssignedEmailNotification(
+          this.formatMessage,
+          theCase,
+          `${this.config.clientUrl}${DEFENDER_ROUTE}/${theCase.id}`,
+        )
+
+        promises.push(
+          this.sendEmail(subject, body, defenderName, defenderEmail),
+        )
+      })
+    } else {
+      const { defenderEmail, defenderNationalId, defenderName } = theCase
+
+      if (!defenderNationalId || !defenderEmail) {
```

Comment:
Same comment as above. I realise that the previous implementation did not send an email to defenders without national ids, but I think that was incorrect.